### DISCUSSION TITLE
key-down--R  Force Reload of Current Source does not appear functional

### DISCUSSION THREAD

**Discussion by StirlingGoetz** [AcademySoftwareFoundation/OpenRV#489] [Q&A] [answered]
Firstly, thank you OpenRV developer(s) for all your hard work. Much appreciated.

My workflow includes using OpenRV 2.0.0 on Windows to read a directory where frame images are being continuously rendered and I need to reopen the sequence regularly to include newly rendered frame images. I am attempting this with the Force Reload of Current Source using hotkey key-down--R which I am assuming is shift--R. 

That hotkey does not appear to reload the source sequence from disk. What am I doing wrong? Is there an automated way to have OpenRV automatically open the new frames that did not exist when I originally opened the sequence?

**Comment by bernie-laberge** [comment]
Hello @StirlingGoetz,

It looks like it the force reload (the reload() RV command which I just realized is not documented!) might not do what you want:


reload():
Reload the frame cache. If no arguments are given, the entire frame
cache will be reloaded. If a global startFrame and endFrame are given,
only frames between them are reloaded.  For sequences, frames that didn't exist originally, but 
exist now and are part of the target range, will be loaded.

https://github.com/AcademySoftwareFoundation/OpenRV/blob/a2bfba4ad37714cdb4a0b7f5e5e30a6a05e5df8d/src/lib/app/mu_rvui/commands.mud#L398

But it looks like the loadChangedFrames() RV command might do the trick:
loadChangedFrames():
For each in the given list of RVFileSource nodes, cached frames that have
changed on disk since they were last read will be reloaded (including the
current frame).  Only works with Sources that hold a sequence of image files.

It should be tied to control C key:
            {"Force Reload Current Frame", ~reload, "R", sourcesExistState},
            {"Force Reload Region", ~reloadInOut, "control R", sourcesExistState},
            {"Reload Changed Frames", ~loadCurrentSourcesChangedFrames, "control C", sourcesExistState},

Note that loadCurrentSourcesChangedFrames() calls loadChangedFrames().
https://github.com/AcademySoftwareFoundation/OpenRV/blob/a2bfba4ad37714cdb4a0b7f5e5e30a6a05e5df8d/src/lib/app/mu_rvui/rvui.mu#L6176

Note that if this is still not doing what you expect it to do, then I would log a new Issue/Bug with all the repro steps so that we can look into it. Thanks!

I hope this helps

**Reply by StirlingGoetz** [reply]
Thanks for the quick response. 

R, Ctrl-R, and Ctrl-C do not appear to reload the image sequence, including newly created image files in the range. From the description above, it sounds like the Ctrl-C might reload changed frames (e.g. existing frames on disk that have changed) when what I am looking for OpenRV to do is to find and load newly rendered frames in the range. Essentially, the equivalent of the Ctrl-R Reload feature in DJV.

**Reply by StirlingGoetz** [reply, marked as answer]
Control-Shift-R and Control-Shift-C do reload the image sequence including new frames, but only if the range was originally opened with the total expected frames. This makes sense despite being less user friendly than with other playback tools.

If the existing image sequence name in the File -> Open dialog is opened it restricts the range to the currently rendered frames and reload hotkeys do nothing. For example, an image sequence with 2524 frames that has only rendered 2450 frames will default in the File -> Open dialog to SequenceName_1-2450#.png and neither hotkey will pick up on the fact that newly rendered frames exist. However, if the image sequence is opened with the name SequenceName_1-2524#.png and reload hotkeys are later used, they do work.
